packages/client/hooks/useMenu.ts

```diff
@@ -53,11 +65,28 @@ const useMenu = <T extends HTMLElement = HTMLButtonElement>(
     return Math.max(40, bbox ? bbox.width : 40)
   }, [originRef.current /* eslint-disable-line react-hooks/exhaustive-deps */])
   const {loadingDelay, loadingDelayRef} = useLoadingDelay()
+
+  const [topOffset, setTopOffset] = useState<number>(0)
+
+  const menuCoords = useMemo(() => {
+    if (
+      [PortalStatus.Mounted, PortalStatus.Entering, PortalStatus.Entered].includes(portalStatus)
+    ) {
+      setTopOffset((cardContainerRef?.current?.scrollTop || 0) - TOP_POSITION_EXCESS)
+    }
+    return 'top' in coords
+      ? {
+          ...coords,
+          top: coords.top + topOffset
+        }
+      : coords
```

Comment:
-1 it's better to refactor as a `guard clauses`, ref: https://refactoring.guru/replace-nested-conditional-with-guard-clauses